### PR TITLE
Add endpoints for the Site object

### PR DESCRIPTION
**Pull request by terrywbrady**
Adding support to obtain the Site object from the repository since this will likely be needed to generate the collection hierarchy.

### CODE REVIEW

**Comment by terrywbrady** [comment]
Note that AIP import/export functions require a user to have access to the site object. If we intend to pass those operations through REST, we will need to expose the site object.

**Comment by terrywbrady** [comment]
Updated base to point to master

File: dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/SiteRestRepository.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+@Component(SiteRest.NAME)
+public class SiteRestRepository extends DSpaceRestRepository<SiteRest, UUID> {
+	SiteService sitesv = ContentServiceFactory.getInstance().getSiteService();
+	@Autowired
+	SiteConverter converter;
+	
+	
+	public SiteRestRepository() {
+		System.out.println("Repository initialized by Spring");
```
Comment (mwoodiupui) [inline dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/SiteRestRepository.java]: Shouldn't this go to the log, at DEBUG level?

**Review by abollini — CHANGES_REQUESTED** [review]

File: dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/SiteRestRepository.java (hunk 1 above)
Comment (abollini) [inline dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/SiteRestRepository.java]: this is a sysout that I have introduced in early stage of the development and we have copied and paste over all the repository now. We should just remove this statement.

**Comment by terrywbrady** [comment]
@abollini , I made the changes requested and have retested this change

**Review by abollini — APPROVED** [review]

**Pull request merged**